backend/coins/eth/account.go

```diff
@@ -446,8 +454,12 @@ func (account *Account) Close() {
 		}
 		account.log.Info("Closed DB")
 	}
+	unlock := account.quitChanLock.Lock()
 	close(account.quitChan)
+	unlock()
+	unlock = account.quitChanLock.Lock()
 	account.quitChan = nil
+	unlock()
```

Comment:
This looks really wrong..but we have to let readers notice the channel closed before we set it to `nil` this is why I lock -> close -> unlock -> lock -> nil.